apps/src/lib/node/ledger/mod.rs

```diff
@@ -340,29 +342,28 @@ async fn run_aux(config: config::Ledger, wasm_dir: PathBuf) {
             res
         });
 
-        let oracle = {
-            #[cfg(not(feature = "eth-fullnode"))]
-            if config.ethereum.oracle_event_endpoint {
+        let oracle = match config.ethereum.mode {
+            ethereum::Mode::Managed => ethereum_node::oracle::run_oracle(
+                ethereum_url,
+                eth_sender,
+                abort_sender,
+            ),
+            ethereum::Mode::EventsEndpoint => {
                 ethereum_node::test_tools::event_endpoint::start_oracle(
                     eth_sender,
                 )
-            } else {
+            }
+            ethereum::Mode::Off => {
                 ethereum_node::test_tools::mock_oracle::run_oracle(
                     ethereum_url,
                     eth_sender,
                     abort_sender,
                 )
```

Comment:
this should only be on if `testing` or `dev` are enabled, no?

Reply:
Re. discussing before, I was initially thinking the Ethereum event endpoint mode could be guarded behind our current `dev` feature as in production, you should never want to use this. So it could be safer to distribute binaries without that functionality to prevent accidental use. That said, I don't think it's worth it. At least currently, the `dev` flag changes multiple things all at once to make local development easier (e.g. where we look for wasms) but there is no fine grained control so putting this mode behind that flag right now means having to turn on that other functionality. Also, the fake Ethereum event endpoint will be most useful to have in CI e2e tests, which use production binaries.

Separately from the eth bridge/this PR, I think the functionality behind the `dev` flag should be split out into separate runtime option(s) rather than hidden behind one compilation flag. It's a bigger discussion though. For now I'd prefer to keep just the one set of options in `eth-bridge-integration` without anything hidden behind a feature, but we should revisit how to prevent accidental misuse in production.